stdlib/_interpchannels.pyi

```diff
@@ -0,0 +1,86 @@
+from _typeshed import structseq
+from typing import Any, Final, Literal, TypeVar, final
+from typing_extensions import Buffer, Self, TypeAlias
+
+_T = TypeVar("_T")
+_SendType: TypeAlias = Literal["send", "recv", "both"]
+
+class ChannelError(RuntimeError): ...
+class ChannelClosedError(ChannelError): ...
+class ChannelEmptyError(ChannelError): ...
+class ChannelNotEmptyError(ChannelError): ...
+class ChannelNotFoundError(ChannelError): ...
+
+@final
+class ChannelID:
+    @property
+    def end(self) -> _SendType: ...
+    @property
+    def send(self) -> Self: ...
+    @property
+    def recv(self) -> Self: ...
+    def __eq__(self, other: object) -> bool: ...
+    def __ge__(self, other: object) -> bool: ...
+    def __gt__(self, other: object) -> bool: ...
+    def __hash__(self) -> int: ...
+    def __index__(self) -> int: ...
+    def __int__(self) -> int: ...
+    def __le__(self, other: object) -> bool: ...
+    def __lt__(self, other: object) -> bool: ...
```

Comment:
It seems like the `<`, `<=`, `>` and `>=` operators are only supported if the object on the other side of the operator is also a `ChannelID` instance:

```pycon
>>> import _interpchannels as _channels
>>> cid = _channels._channel_id(10, force=True)
>>> cid
ChannelID(10)
>>> cid < 'foo'
Traceback (most recent call last):
  File "<python-input-10>", line 1, in <module>
    cid < 'foo'
TypeError: '<' not supported between instances of '_interpchannels.ChannelID' and 'str'
>>> cid < 500
Traceback (most recent call last):
  File "<python-input-11>", line 1, in <module>
    cid < 500
TypeError: '<' not supported between instances of '_interpchannels.ChannelID' and 'int'
```

```suggestion
    def __ge__(self, other: ChannelID) -> bool: ...
    def __gt__(self, other: ChannelID) -> bool: ...
    def __hash__(self) -> int: ...
    def __index__(self) -> int: ...
    def __int__(self) -> int: ...
    def __le__(self, other: ChannelID) -> bool: ...
    def __lt__(self, other: ChannelID) -> bool: ...
```

Reply:
I agree it's largely theoretical given that you can't instantiate the class directly, but I'd prefer to use `ChannelID` as the annotation here. I think it's strictly more accurate, and seeing `Self` used for a comparison dunder like this is quite surprising to me.

Reply:
https://github.com/python/typeshed/pull/12199/commits/6a42e2478a52666fdefb6159e6da99414b49f70d

Reply:
thanks!